### PR TITLE
First draft of an (almost) universal project template

### PR DESCRIPTION
**Pull request by mstade**
This work represents a first stab at a template for open source software projects under the Zambezi banner. It adds sensible defaults for most things, and elsewhere specifies where things should change.

The rationale can probably further explain _why_ some choices were made (such as why MIT, for instance) but other than that this is probably a pretty complete collection of files.

Note that this does not include notably specific files such as project.json, .gitignore, and others. We may want to add those, but we may not, and just create forks of this project, that are effectively variants. (E.g. for a node project, it might be zambezi/project-node, or some such.)

Demo: https://github.com/zambezi/project/tree/v1

### CODE REVIEW

**Comment by mstade** [comment]
@zambezi/contributors I'd love to have your input on this PR, to see if there's anything missing, or whether this need fixing. I think it's pretty extensive as a first cut of a foundation for all our open source projects, but undoubtedly it can improve.

**Comment by mstade** [comment]
We should probably consider adding [issue and PR templates as well](https://github.com/blog/2111-issue-and-pull-request-templates).

File: README.md [hunk 1]
```diff
@@ -1,2 +1,70 @@
...
+Whenever you're starting a project, there are about a million questions that need to be answered – some of them inevitable end up having to be answered for every project. This template attempts to answer the most common ones, providing a solid foundation for you to build on.
+
+Usage
+-----
+
+Do *not* fork this project, unless you intend to actually [contribute changes](CONTRIBUTING.md) which is more than welcome!
+
+To create a new project:
+
+1. Clone (or download) this project: `git clone git@github.com:zambezi/project.git your-project`
```
Comment (matthewdunsdon) [inline README.md]: Could we produce a GitHub release of this project (once PR merged) with a zip attached that contains the minimal files for getting started?

With a zip, the instructions for getting started could be simplified (no .git directory deleting).

Comment (mstade) [inline README.md]: You're right, and when you're right, you're right! :o)

Updated to _much_ simpler instructions in d7ac692.

**Comment by gabrielmontagne** [comment]
@mstade , I like this a lot.  Even though there's no code here, it already feels like a mature project just from the format and the tone  :+1:

**Comment by mstade** [comment]
Awesome, that's what I was going for – thanks @gabrielmontagne! :o)

**Comment by mstade** [comment]
Unless there's any more feedback on this, I'll merge and release v1.0.0 and start overlaying this on the other repos in the org.

**Comment by FabienDeshayes** [comment]
Yeah it's all good stuff!

**Pull request merged**